lib/archethic_web/controllers/api/transaction_controller.ex

```diff
@@ -119,4 +121,37 @@ defmodule ArchEthicWeb.API.TransactionController do
         |> render("400.json", changeset: changeset)
     end
   end
+
+  def origin_public_key_verify(conn, data) do
+    case OriginPublicKeyPayload.changeset(data) do
+      %{valid?: true} ->
+        cert =
+          data["PublicKey"]
+          |> Base.decode16()
+          |> elem(1)
+          |> Crypto.get_key_certificate()
+
+        if cert == data["Certificate"] do
```

Comment:
I think you should use:
- `Crypto.get_root_ca_public_key` to get the root CA from the given public key,
- `Crypto.verify_key_certificate?` to check the given certificate for the given public key and retrieved root CA